src/Microsoft.Framework.PackageManager/Restore/NuGet/HttpSource.cs

```diff
@@ -181,26 +181,35 @@ internal class HttpSource
             return result;
         }
 
-        private async Task<HttpSourceResult> TryCache(string uri, string cacheKey, TimeSpan cacheAgeLimit)
+        internal async Task InvalidateCacheFileAsync(string cacheKey)
         {
-            var baseFolderName = RemoveInvalidFileNameChars(ComputeHash(_baseUri));
-            var baseFileName = RemoveInvalidFileNameChars(cacheKey) + ".dat";
+            var cacheFile = GetCacheFilePath(cacheKey);
 
-#if DNX451
-            var localAppDataFolder = Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData);
-#else
-            var localAppDataFolder = Environment.GetEnvironmentVariable("LocalAppData");
-#endif
-            var cacheFolder = Path.Combine(localAppDataFolder, "dnu", "cache", baseFolderName);
-            var cacheFile = Path.Combine(cacheFolder, baseFileName);
+            // Acquire the lock on a file before we delete it to prevent this process
+            // from deleting a file that just passed existence check in some other processes
+            await ConcurrencyUtilities.ExecuteWithFileLocked(cacheFile, async _ =>
+            {
+                if (File.Exists(cacheFile))
+                {
+                    await Task.Factory.StartNew(() => File.Delete(cacheFile));
```

Comment:
No, just because we need at least one `await` in an async block to get rid of warning. We can also remove the await here and `return Task.FromResult(0)` below. But there is no big difference I guess.